packages/components/atoms/f-button/src/components/Button.vue

```diff
@@ -14,19 +14,36 @@
         v-bind="$attrs"
         :aria-live="getAriaLive"
         :aria-busy="isLoading"
-        v-on="!isLoading && $listeners"
-    >
+        v-on="!isLoading && $listeners">
         <span
             v-if="isLoading"
             :class="$style['c-spinner']"
             :data-test-id="`${componentType}-spinner`" />
-
         <span
-            :class="[
-                $style['o-btn-text'],
-                (isLoading ? $style['o-btn-text--hidden'] : '')
-            ]">
-            <slot />
+            :class="$style['o-button-content']">
+
+            <span
+                v-if="hasIcon && hasIcon === 'leading'"
+                :class="[{
+                    [$style['o-btn-icon']]: hasIcon === 'leading',
+                    [$style['o-btn-icon--leading']]: hasIcon === 'leading' }]">
+                <slot name='leading-icon' />
+            </span>
+
+            <span
+                :class="[$style['o-btn-text'], {
+                    [$style['o-btn-text--hidden']]: isLoading
+                }]">
+                <slot />
+            </span>
+
+            <span
+                v-if="hasIcon && hasIcon === 'trailing'"
+                :class="[{
+                    [$style['o-btn-icon']]: hasIcon === 'trailing',
```

Comment:
Same as the last comment, `hasIcon === 'trailing'` should be split out into a computed property.